### PR TITLE
feat: create product endpoint for superadmin

### PR DESCRIPTION
**Pull request by Oseni03**
#### **Description:**
This pull request introduces a new endpoint for creating products in the application. The endpoint includes the following key features:

1. **Product Creation:**
   - Allows users to create new products by providing necessary details like product name, description, price, category, quantity, image URL, and status.
   - Products are associated with an organization, and the user must belong to this organization to create products.

2. **Authentication:**
   - Users must be authenticated to access the product creation endpoint.
   - Authentication is handled through bearer tokens, obtained via the login endpoint.

3. **Validation:**
   - Validates that all required fields (e.g., `name`, `price`, `org_id`, `category_id`) are present in the request payload.
   - Validates the provided `category_id` against existing categories to ensure it is valid.
   - Returns appropriate HTTP status codes (e.g., `401` for unauthorized, `422` for validation errors, `404` for invalid category).

4. **Testing:**
   - Added unit tests to ensure the product creation endpoint works as expected.
   - Tests cover various scenarios including successful product creation, missing fields, invalid category, and unauthorized access.
   - Mock services and dependencies are used in tests to isolate the functionality of the product creation endpoint.

#### **Key Changes:**
- **New Endpoint**: `POST /api/v1/products/`
- **Model Updates**: Added necessary models and services for handling product creation and validation.
- **Service Updates**: Integrated product category and organization validation within the product creation service.
- **Test Suite**: Expanded test suite with scenarios to validate the correctness and robustness of the endpoint.

#### **Related Issues:**
- [Issue 623](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/623)

#### **Review Notes:**
- Ensure that the endpoint correctly handles all edge cases, particularly around authentication and validation.
- Review the test cases to confirm they cover all relevant scenarios.
- Validate that the organization and category associations are correctly enforced.

#### **Checklist:**
- [x] Endpoint implemented with authentication.
- [x] Input validation and error handling added.
- [x] Unit tests written and passing.
- [x] Documentation updated (if applicable).

### CODE REVIEW

**Comment by joboy-dev** [comment]
Rename the PR since this is specific to auper admin

File: api/v1/routes/product.py [hunk 1]
```diff
@@ -175,3 +190,42 @@ def delete_product(
     """
 
     product_service.delete(db=db, id=id, current_user=current_user)
+
+
+@product.post("/", status_code=status.HTTP_201_CREATED, response_model=success_response)
+async def create_product(
+    product_data: ProductCreate,
+    current_user: Annotated[User, Depends(get_current_user)],
```
Comment (joboy-dev) [inline api/v1/routes/product.py:198]: Change this to user_service.get_current_super_admin since u was told that the endpoint is meant for superadmins alone

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes
